### PR TITLE
Document composition behavior for executable directives

### PR DESCRIPTION
**Pull request by dbanty**

### CODE REVIEW

**Comment by dbanty** [comment]
@clenfest will you also take a look at this and verify that what I say is true? (I can't add reviewers)

File: docs/source/federated-types/composition.mdx [hunk 1]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
 Therefore, the definition of the `Color` enum **must match exactly** in every subgraph that defines it. An exact match is the only scenario that enables union _and_ intersection to produce the same result.
 
 The subgraph schemas above _do not compose_, because their definitions of the `Color` enum differ.
+
+## Directives
+
+Composition will handle a directive differently depending on whether it is an "executable" directive or a "type system" directive.
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:515]:
```suggestion
Composition handles a directive differently depending on whether it's an "executable" directive or a "type system" directive.
```

File: docs/source/federated-types/composition.mdx [hunk 2]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+
+### Executable Directives
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:517]:
```suggestion
### Executable directives
```

File: docs/source/federated-types/composition.mdx [hunk 3]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+
+Executable directives are intended to be used by the client in a query. They are defined on one or more of the [executable directive locations](http://spec.graphql.org/June2018/#ExecutableDirectiveLocation). For example, you might have a directive definition of `directive @lowercase on FIELD` which a client could use in their query like this:
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:519]:
```suggestion
Executable directives are intended to be used by clients in their queries. They are applied to one or more of the [executable directive locations](http://spec.graphql.org/June2018/#ExecutableDirectiveLocation). For example, you might have a directive definition of `directive @lowercase on FIELD`, which a client could use in their query like so:
```

File: docs/source/federated-types/composition.mdx [hunk 4]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+    someField @lowercase
+  }
+}
+```
+
+These types of directives are composed into the supergraph schema only if _all_ of these conditions are met:
+
+- The directive is defined in _all_ subgraphs
+- The directive is defined in the same way in _all_ subgraphs
+- The directive _is not_ used in any [`@composeDirective`](./federated-directives#composedirective) directives
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:533]:
```suggestion
- The directive is defined in _all_ subgraphs.
- The directive is defined _identically_ in _ all subgraphs.
- The directive is _not_ included in any [`@composeDirective`](./federated-directives#composedirective) directives.
```

File: docs/source/federated-types/composition.mdx [hunk 5]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+
+```graphql
+query {
+  getSomeData {
+    someField @lowercase
+  }
+}
+```
+
+These types of directives are composed into the supergraph schema only if _all_ of these conditions are met:
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:529]:
```suggestion
An executable directive is composed into the supergraph schema only if _all_ of the following conditions are met:
```

File: docs/source/federated-types/composition.mdx [hunk 6]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+
+- The directive is defined in _all_ subgraphs
+- The directive is defined in the same way in _all_ subgraphs
+- The directive _is not_ used in any [`@composeDirective`](./federated-directives#composedirective) directives
+
+### Type System Directives
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:535]:
```suggestion
### Type system directives
```

File: docs/source/federated-types/composition.mdx [hunk 7]
```diff
@@ -509,3 +509,29 @@ In this case, the `Color` enum is used as _both_:
...
+
+Type system directives help define the structure of the schema and are not intended for use by clients. They include any directive which is defined on one or more of the [type system directive locations](http://spec.graphql.org/June2018/#TypeSystemDirectiveLocation) These directives will not be composed into the supergraph schema, but can be used to provide information to the router via [`@composeDirective`](./federated-directives#composedirective) directive.
```
Comment (StephenBarlow) [inline docs/source/federated-types/composition.mdx:537]:
```suggestion
Type system directives help define the structure of the schema and are not intended for use by clients. They are applied to one or more of the [type system directive locations](http://spec.graphql.org/June2018/#TypeSystemDirectiveLocation).

These directives are _not_ composed into the supergraph schema, but they can still provide information to the router via the [`@composeDirective`](./federated-directives#composedirective) directive.
```

**Review by StephenBarlow — APPROVED** [review]
Approved with suggstions!

File: docs/source/federated-types/composition.mdx (hunk 4 above)
Comment (dbanty) [reply docs/source/federated-types/composition.mdx:533]: Should `all` still be italicized or no?

Comment (StephenBarlow) [reply docs/source/federated-types/composition.mdx:533]: ah whoops, nah, just "identically" here, now that "all subgraphs" has been emphatically established in the previous bullet

**Pull request merged**